apphub/semantic_segmentation/unet3d_3plus/unet3d_3plus_tf.py

```diff
@@ -0,0 +1,246 @@
+    pool3 = MaxPooling3D(pool_size=(2, 2, 2))(conv3)
+
+    conv4 = encoderblock(pool3, filters * 8)
+
+    up5_4 = upsampleblock(conv4, upchannels, 2)
+    up5_3 = convblock(conv3, upchannels)
+    down5_2 = downsampleblock(conv2, upchannels, 2)
+    down5_1 = downsampleblock(conv1, upchannels, 4)
+
+    if attention:
+        up5_4 = attention_block(attention_upchannels, decoder_input=up5_4, encoder_input=up5_3)
+        down5_2 = attention_block(attention_upchannels, decoder_input=down5_2, encoder_input=up5_3)
+        down5_1 = attention_block(attention_upchannels, decoder_input=down5_1, encoder_input=up5_3)
+
+    merge5 = concatenate([up5_4, up5_3, down5_2, down5_1], axis=4)
+    conv5 = stdconvblock(merge5, upchannels * 4)
+
+    up6_4 = upsampleblock(conv4, upchannels, 4)
+    up6_3 = upsampleblock(conv5, upchannels, 2)
+    up6_2 = convblock(conv2, upchannels)
+    down6_1 = downsampleblock(conv1, upchannels, 2)
+
+    if attention:
+        up6_4 = attention_block(attention_upchannels, decoder_input=up6_4, encoder_input=up6_2)
+        up6_3 = attention_block(attention_upchannels, decoder_input=up6_3, encoder_input=up6_2)
+        down6_1 = attention_block(attention_upchannels, decoder_input=down6_1, encoder_input=up6_2)
+
+    merge6 = concatenate([up6_4, up6_3, up6_2, down6_1], axis=4)
+    conv6 = stdconvblock(merge6, upchannels * 4)
+
+    up7_4 = upsampleblock(conv4, upchannels, 8)
+    up7_3 = upsampleblock(conv5, upchannels, 4)
+    up7_2 = upsampleblock(conv6, upchannels, 2)
+    conv7_1 = convblock(conv1, upchannels)
+
+    if attention:
+        up7_4 = attention_block(attention_upchannels, decoder_input=up7_4, encoder_input=conv7_1)
+        up7_3 = attention_block(attention_upchannels, decoder_input=up7_3, encoder_input=conv7_1)
+        up7_2 = attention_block(attention_upchannels, decoder_input=up7_2, encoder_input=conv7_1)
+    merge7 = concatenate([up7_4, up7_3, up7_2, conv7_1], axis=4)
+    conv7 = stdconvblock(merge7, upchannels * 4)
+
+    conv8 = BatchNormalization(axis=4)(conv7)
+    conv8 = ReLU()(conv8)
+    conv8 = Conv3D(output_classes, 1, activation='sigmoid')(conv8)
+    model = Model(inputs=inputs, outputs=conv8)
+    return model
+
+
+def attention_block(n_filters: int, decoder_input, encoder_input):
+    """An attention unit for Attention Unet 3d.
+
+    Args:
+        n_filters: How many filters for the convolution layer.
+        decoder_input: Input tensor in the decoder section.
+        encoder_input: Input tensor in the encoder section.
+
+    Return:
+        Output Keras tensor.
+    """
+    c1 = Conv3D(n_filters, kernel_size=1)(decoder_input)
+    c1 = BatchNormalization()(c1)
+    x1 = Conv3D(n_filters, kernel_size=1)(encoder_input)
+    x1 = BatchNormalization()(x1)
+    att = ReLU()(x1 + c1)
+    att = Conv3D(1, kernel_size=1)(att)
+    att = BatchNormalization()(att)
+    att = sigmoid(att)
+    return encoder_input * att
+
+
+class ClassEncoding(NumpyOp):
```

Comment:
Don't we have our own Onehot numpyOp?

Reply:
Our current onehot numpyop supports only 1d array.